src/core/geometry/qgsgeometry.cpp

```diff
@@ -836,8 +836,8 @@ QgsGeometry::OperationResult QgsGeometry::splitGeometry( const QgsPointSequence
 
   if ( result == QgsGeometryEngine::Success )
   {
-    *this = newGeoms.takeAt( 0 );
-
+    if ( type() == QgsWkbTypes::PolygonGeometry && ! newGeoms.isEmpty() )
```

Comment:
why doesn't this apply to linestrings? it looks confusing to me

Reply:
No worries. It's really a WIP. I tried something with Travis (I don't always get the same results on my machine).

I think it's cleaner with my last commit. But I'm trouble to understand this fix...